### DISCUSSION TITLE
AWS create bucket returns error statust code 400 IllegalLocationConstraintException

### DISCUSSION THREAD

**Discussion by mordechaihorowitz** [aws/aws-sdk-php#2695] [Q&A] [answered]
status code 400 IllegalLocationConstraintException The unspecified location constraint is incompatible for the region specific endpoint this request was sent to.

Using SDK PHP I am setting the LocationConstraint 

  <code> $response = $s3Client->createBucket(['Bucket' => $bucket_name,
                                           'CreateBucketConfiguration' => [
                                                'LocationConstraint' => 'eu-west-2']
                                         ]);</code>
                                         
 My account is setup for eu-west-2. I am able to retrieve from an existing bucket setup through the console using this location.

The more I look at the logs it appears to me that while my code appears to be setting the LocationConstration I do not believe it is being sent in the request. However I cannot figure out what I am doing wrong or perhaps is it possible this is an issue with the SDK and my code is correct

I have tested with known duplicated bucket name and unique names and received the same error. I have also tried without setting location constraint.  

I have verified through the amazon console I can create a bucket there.
My version in 2.83.  I do want to upgrade but would like to know if ths issue can be fixed without upgrading

**Comment by yenfryherrerafeliz** [comment, marked as answer]
Hi @mordechaihorowitz, this has been answered on [this](https://github.com/aws/aws-sdk-php/issues/2696#issuecomment-1557625429) comment. I wan to add that, if the LocationConstraint parameter is not present, in your request/response, is because you are using an old SDK version, and this can not be fixed unless you upgrade.

Thanks!

[a comment by an automated account is not shown]
